model/riscv_insts_vext_arith.sail

```diff
@@ -0,0 +1,531 @@
+/* ******************************************************************************* */
+/* This file implements part of the vector extension.                              */
+/* Chapter 11: vector integer arithmetic instructions                              */
+/* Chapter 12: vector fixed-point arithmetic instructions                          */
+/* Chapter 16: vector permutation instructions                                     */
+
+/* ******************************************************************************* */
+
+/* **************************OPIVV(VVTYPE)**************************************** */
+
+union clause ast = VVTYPE : (vvfunct6, bits(1), regidx, regidx, regidx)
+
+mapping encdec_vvfunct6 : vvfunct6 <-> bits(6) = {
+  VV_VADD          <-> 0b000000,
+  VV_VSUB          <-> 0b000010,
+  VV_VMINU         <-> 0b000100,
+  VV_VMIN          <-> 0b000101,
+  VV_VMAXU         <-> 0b000110,
+  VV_VMAX          <-> 0b000111,
+  VV_VAND          <-> 0b001001,
+  VV_VOR           <-> 0b001010,
+  VV_VXOR          <-> 0b001011,
+  VV_VRGATHER      <-> 0b001100,
+  VV_VRGATHEREI16  <-> 0b001110,
+  VV_VSADDU        <-> 0b100000,
+  VV_VSADD         <-> 0b100001,
+  VV_VSSUBU        <-> 0b100010,
+  VV_VSSUB         <-> 0b100011,  
+  VV_VSLL          <-> 0b100101,
+  VV_VSMUL         <-> 0b100111,
+  VV_VSRL          <-> 0b101000,
+  VV_VSRA          <-> 0b101001,
+  VV_VSSRL         <-> 0b101010,
+  VV_VSSRA         <-> 0b101011
+}
+
+mapping clause encdec = VVTYPE(funct6, vm, vs2, vs1, vd)
+  <-> encdec_vvfunct6(funct6) @ vm @ vs2 @ vs1 @ 0b000 @ vd @ 0b1010111
+
+function clause execute(VVTYPE(funct6, vm, vs2, vs1, vd)) = {
+  let SEW_pow  : int = get_sew_pow();
+  let SEW      : int = int_power(2, SEW_pow);
+  let LMUL_pow : int = get_lmul_pow();
+  let VLEN_pow : int = get_vlen_pow();
+  let num_elem : int = get_num_elem(LMUL_pow, SEW);
+
+  assert(SEW == 8 | SEW == 16 | SEW == 32 | SEW == 64);
+  let 'n = num_elem;
+  let 'm = SEW;
+
+  let vm_val  : vector('n, dec, bool)     = read_vmask(num_elem, vm, vreg_name("v0"));
+  let vs1_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs1);
+  let vs2_val : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vs2);
+  let vd_val  : vector('n, dec, bits('m)) = read_vreg(num_elem, SEW, LMUL_pow, vd);
+  result      : vector('n, dec, bits('m)) = undefined;
+  mask_helper : vector('n, dec, bool)     = undefined;
```

Comment:
I don't think mask_helper needs to be mutable. And why is it called mask_helper anyway rather than just mask?

Reply:
`result` and `mask_helper` are the return values of `init_masked_result`. We see that `result` needs to be mutable, so we make `mask_helper` mutable too. I'm not sure if it's possible to make `result` mutable while `mask_helper` immutable...

The name of `mask_helper` has been modified as `mask` now.

Reply:
You can always use an immutable intermediate variable

Reply:
When calling the function `(result, mask) = init_masked_result( ... )`, if adding the `let` bound as `let (result, mask) = init_masked_result( ... )`, `result` and `mask` will both be immutable, but the elements of `result` are modified later and cannot be bound by `let`. So I have no idea how to make `mask` bound by `let` while `result` not ...